beacon_node/rest_api/src/beacon.rs

```diff
@@ -493,3 +493,75 @@ pub fn get_genesis_time<T: BeaconChainTypes>(
 ) -> ApiResult {
     ResponseBuilder::new(&req)?.body(&beacon_chain.head()?.beacon_state.genesis_time)
 }
+
+pub fn proposer_slashing<T: BeaconChainTypes>(
+    req: Request<Body>,
+    beacon_chain: Arc<BeaconChain<T>>,
+) -> BoxFut {
+    let response_builder = ResponseBuilder::new(&req);
+
+    let future = req
+        .into_body()
+        .concat2()
+        .map_err(|e| ApiError::ServerError(format!("Unable to get request body: {:?}", e)))
+        .and_then(|chunks| {
+            serde_json::from_slice::<ProposerSlashing>(&chunks).map_err(|e| {
+                ApiError::BadRequest(format!(
+                    "Unable to parse JSON into ProposerSlashing: {:?}",
+                    e
+                ))
+            })
+        })
+        .and_then(move |proposer_slashing| {
+            let spec = &beacon_chain.spec;
+            let state = &beacon_chain.head().unwrap().beacon_state;
+            beacon_chain
+                .op_pool
+                .insert_proposer_slashing(proposer_slashing, state, spec)
+                .map_err(|e| {
+                    ApiError::BadRequest(format!(
+                        "Error while inserting proposer slashing: {:?}",
+                        e
+                    ))
+                })
+        })
+        .and_then(|_| response_builder?.body(&true));
+
+    Box::new(future)
+}
+
+pub fn attester_slashing<T: BeaconChainTypes>(
+    req: Request<Body>,
+    beacon_chain: Arc<BeaconChain<T>>,
+) -> BoxFut {
+    let response_builder = ResponseBuilder::new(&req);
+
+    let future = req
+        .into_body()
+        .concat2()
+        .map_err(|e| ApiError::ServerError(format!("Unable to get request body: {:?}", e)))
+        .and_then(|chunks| {
+            serde_json::from_slice::<AttesterSlashing<T::EthSpec>>(&chunks).map_err(|e| {
+                ApiError::BadRequest(format!(
+                    "Unable to parse JSON into AttesterSlashing: {:?}",
+                    e
+                ))
+            })
+        })
+        .and_then(move |attester_slashing| {
+            let spec = &beacon_chain.spec;
+            let state = &beacon_chain.head().unwrap().beacon_state;
+            beacon_chain
+                .op_pool
+                .insert_attester_slashing(attester_slashing, state, spec)
+                .map_err(|e| {
+                    ApiError::BadRequest(format!(
+                        "Error while inserting attester slashing: {:?}",
+                        e
+                    ))
+                })
+        })
+        .and_then(|_| response_builder?.body(&true));
```

Comment:
`insert_attester_slashing` returns () on success which can't be turned by the ResponseBuilder. We're returning true here, but should we be using something else?